### PR TITLE
🐛 [amp-sidebar] Trigger layout change on toolbar when viewport size changes

### LINKED ISSUE #31042: [amp-sidebar] Toolbar Broken because layout not triggered when viewport size changes
**Please only file reports about bugs in AMP here.**

- If you have questions about how to use AMP or other general questions about AMP please ask them on Stack Overflow under the AMP HTML tag instead of filing an issue here: http://stackoverflow.com/questions/tagged/amp-html
- If you have questions/issues related to Google Search please ask them in Google's AMP forum instead of filing an issue here: https://goo.gl/utQ1KZ

If you have a bug for AMP please fill in the following template. Delete everything except the headers (including this text).

## What's the issue?
[amp-sidebar] Toolbar layout not triggered when viewport size changes, this makes the toolbar to keep showing when it should be hidden or be hidden when it must be shown. Basically the tool bar does not appear when the conditions trigger on viewport change (ex landscape mode, or just chrome window resize)

## How do we reproduce the issue?

https://storywall-76df5.web.app/medical-resources

## What browsers are affected?
All

## Which AMP version is affected?

new issue

### PR DESCRIPTION
**Pull request by dmanek**
Fixed #31042

### CODE REVIEW

**Comment by dmanek** [comment]
Question for @ampproject/wg-performance: the implementation for this issue causes CLS on viewport resize. When I profile it (screenshot below), it's well under 500ms and its marked as "Had recent input: Yes" (as per amp.dev `Layout shifts that occur within 500 milliseconds of user input will have the hadRecentInput flag set, so they can be excluded from calculations.`). Want to confirm this is ok?


cls

**Comment by dmanek** [comment]
Thanks @kevinkimball!

**Comment by kevinkimball** [comment]
cc @ampproject/build-cop for percy

**Comment by caroqliu** [comment]
> cc @ampproject/build-cop for percy

Approved on Percy. Looks like a loader flakiness issue, but should have been fixed by #31124. cc/ @alanorozco

**Comment by dmanek** [comment]
@zhouyx added e2e tests based on offline discussion. Can you PTAL? Thanks.

File: test/fixtures/e2e/amp-sidebar/amp-sidebar-toolbar.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<!doctype html>
+<html ⚡>
+<head>
+    <meta charset="utf-8">
+    <title>amp-sidebar</title>
+    <link rel="canonical" href="//example.com" >
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+    <script async custom-element="amp-sidebar" src="https://cdn.ampproject.org/v0/amp-sidebar-0.1.js"></script>
```
Comment (zhouyx) [inline test/fixtures/e2e/amp-sidebar/amp-sidebar-toolbar.html:10]: Do you also want to add the same test to 0.2 version?

File: extensions/amp-sidebar/0.1/test-e2e/test-amp-sidebar-toolbar.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  async (env) => {
+    let controller;
+
+    beforeEach(() => {
+      controller = env.controller;
+    });
+
+    it('should trigger the toolbar layout when viewport is resized', async () => {
+      await controller.setWindowRect({width: 800, height: 600});
```
Comment (zhouyx) [inline extensions/amp-sidebar/0.1/test-e2e/test-amp-sidebar-toolbar.js:32]: nice!

**Review by zhouyx — APPROVED** [review]
LGTM! Thanks for the fix and the e2e test. One comment

File: test/fixtures/e2e/amp-sidebar/amp-sidebar-toolbar.html (hunk 1 above)
Comment (dmanek) [reply test/fixtures/e2e/amp-sidebar/amp-sidebar-toolbar.html:10]: As discussed offline, e2e tests for amp-sidebar 0.2 are pulling in amp-sidebar-0.1.js. Updating to amp-sidebar-0.2js is causing a bunch of tests to fail. Filed #31233 for tracking this.

**Pull request merged**